nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -7,148 +7,276 @@
+      default = pkgs.xen;
+      defaultText = lib.options.literalExpression "pkgs.xen";
+      example = lib.options.literalExpression "pkgs.xen-slim";
       description = ''
-        The package used for Xen binary.
+        The package used for Xen Hypervisor.
       '';
       relatedPackages = [
         "xen"
-        "xen-light"
+        "xen-slim"
       ];
     };
 
-    virtualisation.xen.package-qemu = mkOption {
-      type = types.package;
-      defaultText = literalExpression "pkgs.xen";
-      example = literalExpression "pkgs.qemu_xen-light";
-      description = ''
-        The package with qemu binaries for dom0 qemu and xendomains.
-      '';
-      relatedPackages = [
-        "xen"
-        {
-          name = "qemu_xen-light";
-          comment = "For use with pkgs.xen-light.";
-        }
-      ];
+    qemu = {
+      package = lib.options.mkOption {
+        type = lib.types.package;
+        default = pkgs.xen;
+        defaultText = lib.options.literalExpression "pkgs.xen";
+        example = lib.options.literalExpression "pkgs.qemu_xen";
+        description = ''
+          The package with QEMU binaries that runs in Domain 0
+          and virtualises the unprivileged domains.
+        '';
+        relatedPackages = [
+          "xen"
+          {
+            name = "qemu_xen";
+            comment = "For use with `pkgs.xen-slim`.";
+          }
+        ];
+      };
+      pidFile = lib.options.mkOption {
+        type = lib.types.path;
+        default = "/run/xen/qemu-dom0.pid";
+        example = "/var/run/xen/qemu-dom0.pid";
+        description = "Path to the QEMU PID file.";
+      };
     };
 
-    virtualisation.xen.bootParams = mkOption {
+    bootParams = lib.options.mkOption {
       default = [ ];
-      type = types.listOf types.str;
-      description = ''
-        Parameters passed to the Xen hypervisor at boot time.
+      example = ''
+        [
+          "iommu=force:true,qinval:true,debug:true"
+          "noreboot=true"
+          "vga=ask"
+        ]
       '';
-    };
-
-    virtualisation.xen.domain0MemorySize = mkOption {
-      default = 0;
-      example = 512;
-      type = types.addCheck types.int (n: n >= 0);
+      type = lib.types.listOf lib.types.str;
       description = ''
-        Amount of memory (in MiB) allocated to Domain 0 on boot.
-        If set to 0, all memory is assigned to Domain 0.
+        Xen Command Line parameters passed to Domain 0 at boot time.
+        Note: these are different from `boot.kernelParams`. See
+        the [Xen documentation](https://xenbits.xenproject.org/docs/unstable/misc/xen-command-line.html) for more information.
       '';
     };
 
-    virtualisation.xen.bridge = {
-      name = mkOption {
-        default = "xenbr0";
-        type = types.str;
+    efi = {
+      bootBuilderFlag = lib.options.mkOption {
+        type = lib.types.enum [
+          "--release"
+          "--report"
+          "--debug"
+          "--quiet"
+        ];
```

Comment:
Since the script is not intended to be called directly (right?), I would prefer to drop the `--`s here and maybe rename it to `bootBuilderVerbosity` or similar. Then the script could also drop the `--`s from its parsing. The fact that it’s passed as a flag feels like a bit of an implementation detail.